src/fps/proposal/IProposal.sol

```diff
@@ -0,0 +1,62 @@
+pragma solidity 0.8.15;
+
+import {AddressRegistry as Addresses} from "src/fps/AddressRegistry.sol";
+
+abstract contract IProposal {
+    /// @notice proposal name, e.g. "BIP15".
+    /// @dev override this to set the proposal name.
+
+    function name() external view virtual returns (string memory);
+
+    /// @notice proposal description.
+    /// @dev override this to set the proposal description.
+    function description() external view virtual returns (string memory);
+
+    /// @notice function to process the proposal to be called from templates.
+    /// @dev use flags to determine which actions to take
+    ///      this function shoudn't be overriden.
+    function _processProposal() internal virtual;
+
+    /// @notice return proposal actions.
+    /// @dev this function shoudn't be overriden.
+    function getProposalActions()
+        external
+        virtual
+        returns (address[] memory targets, uint256[] memory values, bytes[] memory arguments);
+
+    /// @notice return contract identifiers whose storage is modified by the proposal
+    function getAllowedStorageAccess() external view virtual returns (address[] memory);
+
+    /// @notice return proposal calldata
+    function getCalldata() external virtual returns (bytes memory data);
+
+    /// @notice helper function to mock on-chain data
```

Comment:
For my own understanding, what's the purpose of the mock function? Can you explain an example where I might use this? 
I _don't think_ we have this functionality right now in the superchain-ops repo.

Reply:
> we did need to mock some onchain data for one of the tasks around safe modules

I believe that was a misunderstanding of either how to fetch the modules from the safe, or what the expected module address was: https://discord.com/channels/1244729134312198194/1301293771153080331/1326611480585699328

Let's get to the e2e MVP first before we think about what the code and UX should look like for simulating sequences of tasks though

Reply:
Getting rid of the mock functionality for now. If we find that we need it going forward we can add it back in.